### ISSUE TITLE
New scheme is using old code bundle

### ISSUE THREAD

**Issue by gwmccull** [Thinkmill/react-native-schemes-manager#4]
I'm in the process of upgrading our app to RN42 and I came across the issue with custom schemes.

The scheme I'm trying to make is called, `Qa`.  If I create a Qa scheme on each of the libraries that I'm importing, the app compiles and runs with all schemes.

If I use react-native-schemes-manager to create the Qa scheme, the Qa scheme is created in each library (as expected), and the app will compile for all schemes.  

The app will run the latest JavaScript for the Debug and Release schemes but when I try to run the app in the simulator using the Qa scheme, it ends up using an old code bundle.

I bundle my code using:  
```
react-native bundle --platform ios --dev false --entry-file index.ios.js --bundle-output ios-release/main.jsbundle
```

If I run the Qa scheme, I have to manually re-bundle the code after each code change.

If I'm not using react-native-schemes-manager and I just create the Qa schemes manually, it picks up the latest code changes whenever I tell xcode to start the app.

I added some debugging to the node_modules/react-native-schemes-manager/lib/react-native-xcode.sh` script and I get the following inputs:
```bash
DEST=/Users/gmccullough/Library/Developer/Xcode/DerivedData/BuildApp-cznvdybssjczggcpafgzxrjrvgyj/Build/Products/Qa-iphonesimulator/BuildApp.app
DEVELOPMENT_BUILD_CONFIGURATIONS=Qa|Debug
CONFIGURATION=Qa
PLATFORM_NAME=iphonesimulator
Debug build!
Skipping bundling for Simulator platform
```

How can I make it *not* skip the bundling phase?

Looking around the project & target settings I only see one thing that is significantly different for the Qa scheme:
- the target has some Preprocessor Macros set up in the Build Settings
  - the Qa scheme is set to `$(inherited) QA=1`.  This makes for a total inherited value of `DEBUG=1 COCOAPODS=1 QA=1`

Environment:
node: 6.2.2
NPM: 3.9.5
project root folder: BuildApp
iOS project folder: BuildApp/iOS

let me know what other information you need.

Thanks for the help!

### Steps to reproduce the behavior
Install per recommended instructions
Run app using Qa scheme

### Expected behavior
App will run with latest code

### Actual behavior
App runs with old code

**Comment by gwmccull** [comment]
Ok, I figured out the issue a little better.

The Qa scheme that I'm trying to build is actually a clone of the Debug scheme but it requires that the JavaScript code be bundled and packaged into the app.  It doesn't fetch the code from the packager.

The current script, `node_modules/react-native-schemes-manager/lib/react-native-xcode.sh`, assumes that if you're doing a debug scheme, that you'll be using the packager to deliver the code.  Because it assumes this, it exits the script before the bundle step and it doesn't set the DEV flag for the scheme.

So I've come up with a proposed solution:
I've created a new variable in package.json that I'm calling `bundledDebugSchemes`.  `bundledDebugSchemes` is a sibling of the `xcodeSchemes` variable.  It's an optional array of scheme names that should *not* exit the script early.  This will allow the creation of Debug schemes that still use the bundler and set the DEV flag.

I've finished the work but I want to do a little documentation and I'll submit a PR tonight or tomorrow.

**Comment by blargity** [comment]
The skipping of debug schemes when building for the simulator was something that came from React Native originally, so I've been avoiding changing that behaviour unless I have a really good reason to in the future.

What I've actually been doing in my projects is let's say there's, a Dev environment, Staging environment and a Live environment. I might actually want to create a debug build for the Live environment, even though I'd never ship it to customers, so that I can step through and debug behaviour on live data.

So I'm creating:

```
"xcodeSchemes": {
	"Debug": ["Staging-Debug", "Live-Debug"],
	"Release": ["Staging", "Live"]
}
```

And then I'd set something like Jenkins up to only run the Live build, which should definitely ensure the packaging step happens.

On my roadmap for this package is making it so you just put in all the names of the things you want to exist as well as what environment variables you want to exist for each of those, and then it creates both Debug and Release schemes for all of those names and manages pushing the values you put in the config into your code accessible via a native module and managed at build for you.

So basically:

```
"xcodeSchemes": {
	"Debug": {
		serverUrl: "[dev environment URL]"
	},
	"Staging": {
		serverUrl: "[staging environment URL]"
	},
	"Live": {
		serverUrl: "[live environment URL]"
	},
}
```

And then from JS being able to `import` the schemes manager vars and access them (`schemeSettings.serverUrl` in this case). All debug and release configs of the above are created so you can build for debug or release with those configs as you'd like.

Does that fit in with what you'd want to do as well? Or do you have any feedback on this plan in general?

**Comment by gwmccull** [comment]
@blargity Hi, I left a long response on #6 but I wanted to respond here about your roadmap idea.

I like the idea of being able to specify the parameters in the package.json file and then have them available in the code.  One of the big things we use the __DEV__ flag for is to set parameters and API keys.

The documentation would have to be pretty clear about not putting private API keys in a package.json file on a public repo.  It's easy to .gitignore an API keys file but you can't do that to your package.json file.

We also use the __DEV__ flag to enable/disable certain features that we only want for our internal QA team but it seems like we could still rely on that (or using a DEV flag in the `xcodeSchemes` objects

Right now, the package looks at the xcodeSchemes keys so it knows which scheme to duplicate with a new name.  

In your idea, how would you specify which schemes were being duplicated?

**Comment by gwmccull** [comment]
Some of this was fixed in #9 and some was decided not to do
